packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -529,12 +532,12 @@ class TimeInput extends React.Component {
       timeValue = 'T'.concat(hour, ':', this.state.minute);
     }
 
-    if (hourAttributes && !hourAttributes.id) {
-      hourAttributes.id = 'terra-time-hour-'.concat(name);
+    if (instanceHoursAttrs && !instanceHoursAttrs.id) {
```

Comment:
Nitpicking here but you should be able to remove the presence check for instanceHoursAttrs (and instanceMinuteAttrs below).